packages/steps-s3-copy/src/lib/thaw-large-copy-map-construct.ts

```diff
@@ -0,0 +1,71 @@
+import { Construct } from "constructs";
+import { Chain, IChainable } from "aws-cdk-lib/aws-stepfunctions";
+import { ThawObjectsMapConstruct } from "./thaw-objects-map-construct";
+import { CopyMapConstruct } from "./copy-map-construct";
+import { IRole } from "aws-cdk-lib/aws-iam";
+import { SubnetType } from "aws-cdk-lib/aws-ec2";
+import {
+  ICluster,
+  TaskDefinition,
+  ContainerDefinition,
+} from "aws-cdk-lib/aws-ecs";
+
+interface ThawLargeCopyMapProps {
+  readonly writerRole: IRole;
+  readonly workingBucket: string;
+  readonly workingBucketPrefixKey: string;
+  readonly inputPath: string;
+  readonly aggressiveTimes?: boolean;
+
+  readonly cluster: ICluster;
+  readonly clusterVpcSubnetSelection: SubnetType;
+  readonly taskDefinition: TaskDefinition;
+  readonly containerDefinition: ContainerDefinition;
+
+  readonly maxItemsPerBatch: number;
+  readonly maxConcurrency: number;
+}
+/**
+ * Two-step construct that restores cold-storage objects and then transfers them using Fargate tasks.
+ * Composed of ThawObjectsMapConstruct → CopyMapConstruct (from thaw-objects-map-construct and copy-map-construct)
+ */
+export class ThawLargeCopyMapConstruct extends Construct {
+  public readonly chain: IChainable;
+  public readonly distributedMap;
+  public readonly thawStep;
+  public readonly copyStep;
+
+  constructor(scope: Construct, id: string, props: ThawLargeCopyMapProps) {
+    super(scope, id);
+
+    // Step 1: Thaw objects from cold storage using a Lambda-driven distributed map
+    const thawStep = new ThawObjectsMapConstruct(this, "ThawLargeObjects", {
+      writerRole: props.writerRole,
+      workingBucket: props.workingBucket,
+      workingBucketPrefixKey: props.workingBucketPrefixKey,
+      aggressiveTimes: props.aggressiveTimes,
+      inputPath: props.inputPath,
+      mapStateName: id,
+    });
+
```

Comment:
What would be the ramifications if one thaw completed quickly - and the others took 48 hours?

Might the quick thaw "re-expire"??

Reply:
I thought potentially for (large or small) thawing we would have only one distributed map - but that _within_ that map would be a single chain for each object doing the thaw->wait->copy loop